Sources/CardPayments/CardClient.swift

```diff
@@ -7,27 +7,78 @@ import CorePayments
 public class CardClient: NSObject {
 
     public weak var delegate: CardDelegate?
-
+    public weak var vaultDelegate: CardVaultDelegate?
+    
     private let apiClient: APIClient
     private let config: CoreConfig
     private let webAuthenticationSession: WebAuthenticationSession
     private var analyticsService: AnalyticsService?
+    private var graphQLClient: GraphQLClient?
 
     /// Initialize a CardClient to process card payment
     /// - Parameter config: The CoreConfig object
     public init(config: CoreConfig) {
         self.config = config
         self.apiClient = APIClient(coreConfig: config)
         self.webAuthenticationSession = WebAuthenticationSession()
+        self.graphQLClient = GraphQLClient(environment: config.environment)
     }
 
     /// For internal use for testing/mocking purpose
-    init(config: CoreConfig, apiClient: APIClient, webAuthenticationSession: WebAuthenticationSession) {
+    init(
+        config: CoreConfig,
+        apiClient: APIClient,
+        webAuthenticationSession: WebAuthenticationSession,
+        graphQLClient: GraphQLClient? = nil
+    ) {
         self.config = config
         self.apiClient = apiClient
         self.webAuthenticationSession = webAuthenticationSession
+        self.graphQLClient = graphQLClient
     }
+    
+    public func vault(vaultRequest: CardVaultRequest) {
```

Comment:
I know on the BT side for similar methods we moved the requests to be unnamed parameters. How would we feel about doing something like that here:
```suggestion
    public func vault(_ vaultRequest: CardVaultRequest) {
```